stylix/mk-target.nix

```diff
@@ -0,0 +1,161 @@
+# Provides a consistent target interface, minimizing boilerplate and
+# automatically safeguarding declarations related to disabled options.
+#
+# This function returns the module body, not an actual module.
+#
+# The /modules/<MODULE>/<PLATFORM>.nix modules should use this function as
+# follows:
+#
+#     { mkTarget, lib... }:
+#     mkTarget {
+#       name = "example";
+#       humanName = "Example Target";
+#
+#       generalConfig =
+#         lib.mkIf complexCondition {
+#           home.packages = [ pkgs.hello ];
+#         };
+#
+#       configElements = [
+#         { programs.example.theme.name = "stylix"; }
+#
+#         (
+#           { colors }:
+#           {
+#             programs.example.theme.background = colors.base00;
+#           }
+#         )
+#
+#         (
+#           { fonts }:
+#           {
+#             programs.example.font.name = fonts.monospace.name;
+#           }
+#         )
+#       ];
+#     }
+#
+# Note that `mkTarget` is provided in the same place as module arguments.
+#
+# Ideally, in the future, this function also prevents unguarded module arguments
+# by limiting the arguments passed through `/stylix/autoload.nix`
```

Comment:
Is this comment more appropriate in autoload.nix? Since that is where the restriction will be added, not here.

Reply:
> Is this comment more appropriate in autoload.nix? Since that is where the restriction will be added, not here.

I placed this TODO comment in the `mkTarget` documentation because it applies to `mkTarget`. This avoids the general scalability problem where the general consumer (`/stylix/autoload.nix`) is littered with TODO comments beyond what the consumer should care about. Although this scalability issue will likely never happen in Stylix, this is the general reasoning behind why I put it in the `mkTarget` documentation.